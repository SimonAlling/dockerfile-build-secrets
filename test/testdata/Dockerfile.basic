FROM alpine

RUN --mount=type=secret,id=foo echo

RUN --mount=type=secret,id=bar echo
