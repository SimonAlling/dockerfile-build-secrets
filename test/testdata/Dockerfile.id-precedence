FROM alpine

# Only two of id, source, target specified:
RUN --mount=type=secret,id=foo-id,source=foo-source echo
RUN --mount=type=secret,id=bar-id,target=/run/bar-target echo
RUN --mount=type=secret,source=baz-source,target=/run/baz-target echo

# All three specified:
RUN --mount=type=secret,id=quux-id,source=quux-source,target=/run/quux-target echo
