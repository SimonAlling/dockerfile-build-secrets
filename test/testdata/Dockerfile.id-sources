FROM alpine

RUN --mount=type=secret,id=foo echo

RUN --mount=type=secret,source=bar echo

RUN --mount=type=secret,target=/run/baz echo
