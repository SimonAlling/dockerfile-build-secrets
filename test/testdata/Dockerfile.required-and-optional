FROM alpine

RUN --mount=type=secret,id=foo --mount=type=secret,id=foo,required=true echo

RUN --mount=type=secret,id=bar,required=true --mount=type=secret,id=bar echo

RUN --mount=type=secret,id=baz,required=false echo
RUN --mount=type=secret,id=baz,required=true echo

RUN --mount=type=secret,id=quux,required=true echo
RUN --mount=type=secret,id=quux,required=false echo
