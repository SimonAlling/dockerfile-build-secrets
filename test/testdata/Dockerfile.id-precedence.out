{"bar-id":{"required":false},"baz-source":{"required":false},"foo-source":{"required":false},"quux-source":{"required":false}}
